File: Queries on a Permutation With Key/Leetcode_1409.py
class Solution:
    def processQueries(self, queries: List[int], m: int) -> List[int]:
        res = []
        P = list(range(1,m+1))
        for q in queries:
            idx= P.index(q)
            res += [idx]
            P = [q] + P[:idx] + P[idx+1:]
        return res

File: Queries on a Permutation With Key/Leetcode_1409.cpp
/*
Time complexity: O(m*n) where n=len(queries)
since all list operations are linear time
*/

class Solution {
public:
    vector<int> processQueries(vector<int>& queries, int m) {
        vector<int> res(queries.size());
        
        list<int> P;
        for (int i=1; i<=m; ++i) P.emplace_back(i);
           
        int idx;
        for (int j=0; j<queries.size(); ++j){
            // cout<<"query: "<<queries[j]<<endl;
            list<int>::iterator it; idx=0;
            for (it=P.begin(); it!=P.end(); it++){
                // cout<<"*it="<<*it<<endl;
                if (*it==queries[j]) {
                    // cout<<"FOUND"<<idx<<endl;
                    res[j] = idx;
                    break;
                }
                idx++;
            }
            P.erase(it);
            P.emplace_front(queries[j]);
        }
        return res;
    }
};
